src/tooltip.ts

```diff
@@ -138,14 +134,29 @@ class CompletionTooltip {
         if (this.info) this.view.requestMeasure(this.placeInfo)
       })
     }
-
     if (this.updateSelectedOption(open.selected)) {
       if (this.info) {this.info.remove(); this.info = null}
-      let option = open.options[open.selected]
-      if (option.completion.info) {
-        this.info = this.dom.appendChild(createInfoDialog(option, this.view)) as HTMLElement
+      let {completion} = open.options[open.selected]
+      let {info} = completion
+      if (!info) return
+      if (typeof info === 'string') {
+        this.info = this.dom.appendChild(createInfoDomNode(info))
         this.view.requestMeasure(this.placeInfo)
+        return;
+      }
+      let infoResult = info(completion)
+      if ('then' in infoResult) {
+        infoResult.then(maybeNode => {
+          if (!maybeNode) return
+          let newState = this.view.state.field(this.stateField)
+          if (newState != cState) return
```

Comment:
Not sure if this is enough to ensure we're in the same context for the completion